contrib/mimalloc-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+if (OS_LINUX AND NOT SANITIZE AND NOT ARCH_ARM AND NOT ARCH_32 AND NOT ARCH_PPC64LE)
+    option (ENABLE_MIMALLOC "Enable mimalloc allocator" ${ENABLE_LIBRARIES})
+endif ()
+
+if (NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/mimalloc/include/mimalloc.h")
+   message (WARNING "submodule contrib/mimalloc is missing. to fix try run: \n git submodule update --init --recursive")
+endif ()
+
+if (ENABLE_MIMALLOC)
+    set (MIMALLOC_INCLUDE_DIR ${ClickHouse_SOURCE_DIR}/contrib/mimalloc/include)
+    set (USE_MIMALLOC 1)
+    set (MIMALLOC_LIBRARY mimalloc-static)
```

Comment:
I don't see `mimalloc-static` anywhere in this cmake rules.

Reply:
I am following guide from [mimalloc/README.md](https://github.com/microsoft/mimalloc#using-the-library). Looks like it is defined somewhere inside `mimalloc` itself.
Furthermore, it is done in the same way in [previous attempt](https://github.com/ClickHouse/ClickHouse/pull/5775/files#diff-94d7241e94911b5fc9c995adb12b3aa127323a9f9709a9d8bb2fbddd5b9c80e0R12) to use `mimalloc`. (#5775)

Reply:
You don't neither `MIMALLOC_LIBRARY`, so you don't need this variable. And AFAICS right now you added only header only library, so it does not matter static or shared.

Also note (for the future), that there is module part too, i.e. not headers only. Seems that it is required for overrides - https://github.com/microsoft/mimalloc#static-override